File: android/app/src/main/java/gg/strims/ppspp/rpc/SwarmPeerClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class SwarmPeerClient(filepath: String) : RPCClient(filepath) {

    suspend fun announceSwarm(
        arg: SwarmPeerAnnounceSwarmRequest = SwarmPeerAnnounceSwarmRequest()
    ): SwarmPeerAnnounceSwarmResponse =
        this.callUnary("SwarmPeer/AnnounceSwarm", arg)

}

File: ios/App/App/API/SwarmPeerClient.swift
// swift-format-ignore-file
//
//  SwarmPeerClient.swift
//  App
//
//  Copyright © 2020 MemeLabs. All rights reserved.
//

import Foundation
import PromiseKit

class SwarmPeerClient {
  private var client: RPCClient

  init(_ client: RPCClient) {
    self.client = client
  }

  public func announceSwarm(_ arg: PBSwarmPeerAnnounceSwarmRequest = PBSwarmPeerAnnounceSwarmRequest()) -> Promise<PBSwarmPeerAnnounceSwarmResponse> {
    return self.client.callUnary("SwarmPeer/AnnounceSwarm", arg)
  }
  
}
